test/proposal/space_maker_test.rb

```diff
@@ -303,57 +303,58 @@ def probed_partition(name)
       # In the past, SpaceMaker used to delete the extended partition sda4
       # leaving sda6 alive. This test ensures the bug does not re-appear
       it "does not delete the extended partition if some logical one is to be reused" do
-        volumes = vols_list(
-          planned_vol(mount_point: "/1", type: :ext4, desired: 400.GiB),
-          planned_vol(mount_point: "/2", reuse: "/dev/sda1"),
-          planned_vol(mount_point: "/3", reuse: "/dev/sda2"),
-          planned_vol(mount_point: "/4", reuse: "/dev/sda3"),
-          planned_vol(mount_point: "/5", reuse: "/dev/sda6")
-        )
+        partition = proposed_partition(mount_point: "/1", type: :ext4, disk_size: 400.GiB)
+        proposed_partitions = [partition]
+        keep = ["/dev/sda1", "/dev/sda2", "/dev/sda3", "/dev/sda6"]
 
-        expect { maker.provide_space(volumes) }.to raise_error Y2Storage::Proposal::NoDiskSpaceError
+        expect { maker.provide_space(proposed_partitions, partitions_to_keep: keep) }.to(
+          raise_error Y2Storage::Proposal::NoDiskSpaceError
+        )
       end
     end
 
     context "when some volumes must be reused" do
       let(:scenario) { "multi-linux-pc" }
-      let(:volumes) do
-        vols_list(
-          planned_vol(mount_point: "/1", type: :ext4, desired: 60.GiB),
-          planned_vol(mount_point: "/2", type: :ext4, desired: 300.GiB),
-          planned_vol(mount_point: "/3", reuse: "/dev/sda6"),
-          planned_vol(mount_point: "/4", reuse: "/dev/sda2")
-        )
+      let(:proposed_partitions) do
+        [
+          proposed_partition(mount_point: "/1", type: :ext4, disk_size: 60.GiB),
+          proposed_partition(mount_point: "/2", type: :ext4, disk_size: 300.GiB)
+        ]
+      end
+      let(:keep) { ["/dev/sda6", "/dev/sda2"] }
+
+      before do
+        skip("Do these tests make sense after adding partitions_to_keep parameter? ")
```

Comment:
I guess many of the tests should be moved to the caller, making sense that it calls `SpaceMaker` with the right `partitions` and `keep` arguments.